src/acquisition/covidcast/csv_importer.py

```diff
@@ -0,0 +1,242 @@
+"""Collects and reads covidcast data from a set of local CSV files."""
+
+# standard library
+import glob
+import os
+import re
+
+# third party
+import pandas
+
+
+class CsvImporter:
+  """Finds and parses covidcast CSV files."""
+
+  # .../source/yyyymmdd_geo_signal.csv
+  PATTERN_DAILY = re.compile(r'^.*/([^/]*)/(\d{8})_(\w+?)_(\w+)\.csv$')
+
+  # .../source/weekly_yyyyww_geo_signal.csv
+  PATTERN_WEEKLY = re.compile(r'^.*/([^/]*)/weekly_(\d{6})_(\w+?)_(\w+)\.csv$')
+
+  # set of allowed resolutions (aka "geo_type")
+  GEOGRAPHIC_RESOLUTIONS = {'county', 'hrr', 'msa', 'dma', 'state'}
+
+  # set of required CSV columns
+  REQUIRED_COLUMNS = {'geo_id', 'val', 'se', 'direction', 'sample_size'}
+
+  # NOTE: this should be a Python 3.7+ `dataclass`, but the server is on 3.4
+  # See https://docs.python.org/3/library/dataclasses.html
+  class RowValues:
+    """A container for the values of a single covidcast row."""
+
+    def __init__(self, geo_value, value, stderr, sample_size, direction):
+      self.geo_value = geo_value
+      self.value = value
+      self.stderr = stderr
+      self.sample_size = sample_size
+      self.direction = direction
+
+  @staticmethod
+  def is_sane_day(value):
+    """Return whether `value` is a sane (maybe not valid) YYYYMMDD date."""
+
+    year, month, day = value // 10000, (value % 10000) // 100, value % 100
+
+    nearby_year = 2014 <= year <= 2023
+    valid_month = 1 <= month <= 12
+    sensible_day = 1 <= day <= 31
+
+    return nearby_year and valid_month and sensible_day
+
+  @staticmethod
+  def is_sane_week(value):
+    """Return whether `value` is a sane (maybe not valid) YYYYWW epiweek."""
+
+    year, week = value // 100, value % 100
+
+    nearby_year = 2014 <= year <= 2023
+    sensible_week = 1 <= week <= 53
+
+    return nearby_year and sensible_week
+
+  @staticmethod
+  def find_csv_files(scan_dir, glob=glob):
+    """Recursively search for and yield covidcast-format CSV files.
+
+    scan_dir: the directory to scan (recursively)
+
+    The return value is a tuple of (path, details), where, if the path was
+    valid, details is a tuple of (source, signal, time_type, geo_type,
+    time_value) (otherwise None).
+    """
+
+    for path in glob.glob(os.path.join(scan_dir, '*', '*')):
+      if not path.lower().endswith('.csv'):
+        # safe to ignore this file
+        continue
+
+      print('file:', path)
+
+      # match a daily or weekly naming pattern
+      daily_match = CsvImporter.PATTERN_DAILY.match(path.lower())
+      weekly_match = CsvImporter.PATTERN_WEEKLY.match(path.lower())
+      if not daily_match and not weekly_match:
+        print(' invalid csv path/filename')
+        yield (path, None)
+        continue
+
+      # extract and validate time resolution
+      if daily_match:
+        time_type = 'day'
+        time_value = int(daily_match.group(2))
+        match = daily_match
+        if not CsvImporter.is_sane_day(time_value):
+          print(' invalid filename day')
```

Comment:
Include the `time_value` parsed here, since if this goes wrong inappropriately that's the first thing we'll want to know

Reply:
done